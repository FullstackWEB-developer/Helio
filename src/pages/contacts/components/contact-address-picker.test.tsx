import i18n from '../../../i18nForTests';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import utc from 'dayjs/plugin/utc';
import customParseFormat from 'dayjs/plugin/customParseFormat';
import {unmountComponentAtNode} from 'react-dom';
import {render} from '@testing-library/react';
import TestWrapper from '@shared/test-utils/test-wrapper';
import ContactAddressPicker from './contact-address-picker';
import Router from "react-router-dom";
jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useParams: jest.fn(),
   }));
describe("ContactAddressPicker tests", () => {
    let container: HTMLDivElement | null;
    const mockState = {
        emailState: {
            unreadEmails: 0,
            messageSummaries: []
        },
        appUserState: {
            liveAgentStatuses: [],
            appUserDetails: {
                id: ""
            }
        },
        lookupsState: {

        },
        ticketState: {
            lookupValues: []
        },
        appState: {
            smsTemplates: [],
            emailTemplates: []
        },
        ccpState: {
            chatCounter: 1,
            voiceCounter: 2
        }
    };

    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        dayjs.extend(utc);
        dayjs.extend(customParseFormat);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders ContactAddressPicker correctly", async () => {
        let option = {
            value: "a",
            label: "a",
        };
        jest.spyOn(Router, 'useParams').mockReturnValue({ id: '1234' })
        const {asFragment} = render(<TestWrapper mockState={mockState}>
            <ContactAddressPicker onSelect={() => {}} options={[option]}/>
        </TestWrapper>);
        expect(asFragment()).toMatchSnapshot();
    });
})
