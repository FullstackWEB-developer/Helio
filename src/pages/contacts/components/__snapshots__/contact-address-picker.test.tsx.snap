// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactAddressPicker tests renders ContactAddressPicker correctly 1`] = `
<DocumentFragment>
  <div
    class="relative py-2.5 cursor-pointer"
  >
    <div
      class="flex items-center"
    >
      <div
        class=""
      >
        <svg
          class="icon-medium "
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
          />
          <path
            class="fill-default"
            d="M19,13H13v6H11V13H5V11h6V5h2v6h6Z"
          />
        </svg>
      </div>
      <span
        class="body2 pl-2 contact-accent-color"
      >
        contacts.new-contact.add_more
      </span>
    </div>
  </div>
</DocumentFragment>
`;
